src/main/java/org/folio/rest/impl/HoldingsRecordsSourceAPI.java

```diff
@@ -0,0 +1,193 @@
+                  .headersFor201().withLocation(LOCATION_PREFIX + ret))));
+              } else {
+                String msg = PgExceptionUtil.badRequestMessage(reply.cause());
+                msg = (msg == null) ? "Internal server problem: Error message missing" : msg;
+                log.info(msg);
+                asyncResultHandler.handle(Future.succeededFuture(PostHoldingsRecordsSourcesResponse
+                  .respond400WithTextPlain(msg)));
+              }
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          asyncResultHandler.handle(Future.succeededFuture(PostHoldingsRecordsSourcesResponse
+            .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+        }
+      });
+  }
+
+  @Override
+  public void getHoldingsRecordsSourcesById(String id, String lang,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+      PgUtil.getById(REFERENCE_TABLE, HoldingsRecordsSource.class, id, okapiHeaders,
+        vertxContext, GetHoldingsRecordsSourcesByIdResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void deleteHoldingsRecordsSourcesById(String id, String lang,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+      vertxContext.runOnContext(v -> {
+        try {
+          String tenantId = TenantTool.tenantId(okapiHeaders);
+          PostgresClient postgres = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+          postgres.delete(REFERENCE_TABLE, id,
+            reply -> {
+              if (reply.failed()) {
+                String msg = PgExceptionUtil.badRequestMessage(reply.cause());
+                msg =  (msg == null) ? "Internal server problem: Error message missing" : msg;
+                log.info(msg);
+                asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+                  .respond400WithTextPlain((msg))));
+                return;
+              }
+              int updated = reply.result().rowCount();
+              if (updated != 1) {
+                String msg = messages.getMessage(lang, MessageConsts.DeletedCountError, 1, updated);
+                log.error(msg);
+                asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+                  .respond404WithTextPlain(msg)));
+                return;
+              }
+              asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+                .respond204()));
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          asyncResultHandler.handle(succeededFuture(DeleteHoldingsRecordsSourcesByIdResponse
+            .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+        }
+      });
```

Comment:
Is there any reason why standard org.folio.rest.persist.PgUtil#deleteById(...) method can not be used?